### DISCUSSION TITLE
Run all rules at once

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [github/codeql#10374] [Q&A] [answered]
I'm looking for something equivalent to:

```
semgrep --config "r/java"
```

Is there any similar commands?

**Comment by MathiasVP** [comment, marked as answer]
Hi @CaledoniaProject,

You can run all queries in a given suite on a database using a command like:
```
codeql database analyze /path/to/my-database java-security-and-quality.qls 
```
which will run all of the queries in the Java security-and-quality suite.

Note that CodeQL will schedule all the queries to be run as soon as the engine determines that it's best to start them. This won't always mean that they all run at once (since it might not be the best option due to resource constraints like memory pressure, number of available cores, etc.)

Does that help?
